### PR TITLE
fix: add services connect timeout

### PR DESCRIPTION
**Pull request by korhaliv**
This pr improves error handling during connect stage and adds a timeout to a base Service class, which ensures that `grpc.connect` always finishes execution. The PR also handles `Lightning` service connection failure properly by closing all its child services.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -0,0 +1,4368 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (mrfelton) [inline yarn.lock]: Remove `yarn.lock`. This package is using npm not yarn.

File: src/utils/promiseTimeout.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/**
+ * promiseTimeout - Reject a promise if it doesn't resolve after a given timeout window.
+ *
+ * @param {number} ms Timeout (ms)
+ * @param {Promise} promise Promise to timeout
```
Comment (mrfelton) [inline src/utils/promiseTimeout.js]: Missing doc for `message` param.

Comment (mrfelton) [reply src/utils/promiseTimeout.js]: Also, I'd suggest making the `message` param come last and optional and provide a default message in the case it's not provided.

Comment (korhaliv) [reply src/utils/promiseTimeout.js]: updated

File: yarn.lock (hunk 1 above)
Comment (korhaliv) [reply yarn.lock]: removed

**Comment by korhaliv** [comment]
@mrfelton ready for retest

**Pull request merged**